### ISSUE TITLE
Clarification about first condition in replay protection implementation

### ISSUE THREAD

**Issue by StefDevs** [networkprotocol/netcode.io#64]
From the provided implementation:
`
int netcode_replay_protection_packet_already_received( struct netcode_replay_protection_t * replay_protection, uint64_t sequence )
{
    netcode_assert( replay_protection );

    if ( sequence & ( 1ULL << 63 ) )
        return 0;
`

If a packet arrives that is within replay buffer size of the most recent sequence number, it is accepted only if its sequence number has not already been received, otherwise it is ignored.

**Comment by gafferongames** [comment]
Yes, this looks weird. I'll take a closer look. It may be historical...

I remember at some point my replay protection code had the high bit set for specific packet types that had sequence numbers, but shouldn't have replay protection applied to them. I'm not sure if this still applies to netcode.io protocol though.

If you disable this check, do the netcode.io tests still pass?

cheers

- Glenn

**Comment by StefDevs** [comment]
Right on, thanks for looking into it. 

To answer your question I'm in C# for Unity, but in a much more "data oriented" (if you wanna call it that) way with intent to move it to jai when Thekla gets around to releasing that. 

- Stef

**Comment by gafferongames** [comment]
Great stuff. Are you planning on sharing your implementation open source? I'd be happy to link to it and publicize it, even if it's a work in progress.

cheers

- Glenn

**Comment by StefDevs** [comment]
I can certainly share it when its done, but I am taking some liberties with the assumption that it's for personal use so it may not be exactly to spec. For example I'm disregarding the seperation of "netcode.io" and reliability. 

I dont think I'm exactly qualified to be releasing libraries but I'll share as an example regardless.

**Comment by gafferongames** [comment]
Ahhhhh this is like what SuperNetcode.io is doing as well in C#...
